### PR TITLE
ROX-17083: Make PagerDuty the default receiver

### PR DESCRIPTION
**Pull request by kylape**
## Description
This is part of a larger effort to stop treating stage alerts as critical alerts in PagerDuty.  The specific changes being made in this PR:

* Pass the `severity` parameter in the PagerDuty receiver configuration. This will be used in production to capture non-critical production events in PagerDuty without triggering a critical notification to on-call engineers.  (These events are currently dropped in Alertmanager)
* Switch to the PagerDuty v2 Events API.  This is required for PD to recognize the `severity` parameter.  This is done by switching from `service_key` to `routing_key`.
* Updated the key in `terraform_cluster.sh` (and thus AWS Secrets Manager) to reflect the routing key change.
* Made PagerDuty the default receiver in the Alertmanager config.  This will send non-critical alerts to PagerDuty, treating it as a hub for all events across all data plane clusters.

It's implied that the PagerDuty routing key for the stage environment will be different from the one in production.  This is so the stage service can be configured to force all incidents to be considered low priority to avoid paging on-call engineers.

## Checklist (Definition of Done)

- [x] Unit and integration tests added
- [x] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [x] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [x] Add secret to app-interface Vault or Secrets Manager if necessary
- [x] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [x] Check AWS limits are reasonable for changes provisioning new resources

## Test manual

* get the `routing_key`
* set the template string in the PD severity `{{ or .GroupLabels.severity 'info' }}`
* render helm chart and get `alertmanager.yaml`
* fire up test flask app (prometheus exporter)
* configure alert to be a non-critical severity
* fire up alertmanager and prometheus locally
* trigger alert with curl command
* this should create an incident in PD and send to slack, but NOT notify me personally
* let it resolve
* configure the alert to be a critical severity
* trigger alert with curl command
* this should page me as well as create a slack message (won't normally happen
  after updating notifcation config in stage PD service)
* let it resolve

### CODE REVIEW

**Comment by kylape** [comment]
I went ahead and added the routing key to the Secret Manager for stage (but not for prod).

File: dp-terraform/helm/rhacs-terraform/charts/observability/templates/01-operator-03-secret-alertmanager.yaml [hunk 1]
```diff
@@ -8,23 +8,19 @@ stringData:
...
         match:
           alertname: DeadMansSwitch
           observability: managed-rhacs
     receivers:
-    - name: default-receiver
     - name: managed-rhacs-pagerduty
       pagerduty_configs:
-      - service_key: {{ .Values.pagerduty.key | quote }}
+      - routing_key: {{ .Values.pagerduty.key | quote }}
+        severity: "{{ "{{" }} or .GroupLabels.severity \"info\" }}"
```
Comment (stehessel) [inline dp-terraform/helm/rhacs-terraform/charts/observability/templates/01-operator-03-secret-alertmanager.yaml:23]: Is this even correct yaml? Looks like some `"` are not correct escaped here? Can you maybe explain what this is supposed to do? How are `info` alerts handled?

**Comment by stehessel** [comment]
I think we have to notify on-call engineers + SRE to set up their high/low urgency notification settings correctly. Otherwise it might have the opposite effect of *more* pages over the weekend, when someone has low urgency linked to their phone and we now sent alerts for `warning`/`info`.

**Comment by kylape** [comment]
That's true for the production environment.  For stage, the schedule is just me, so no one will get paged for those.  For this weekend, I can contact the on-call engineer to have them set up the notifications correctly before the weekend starts.

File: dp-terraform/helm/rhacs-terraform/charts/observability/templates/01-operator-03-secret-alertmanager.yaml (hunk 1 above)
Comment (kylape) [reply dp-terraform/helm/rhacs-terraform/charts/observability/templates/01-operator-03-secret-alertmanager.yaml:23]: > I guess this is what you get when you have merge yaml, Helm and PagerDuty templating
Exactly :)

Honestly I thought the quotes inside the quotes wouldn't work, but it worked when I tested it.  Let me try to explain this line in plain english.

I want the severity to be based on the severity label coming from the alert itself.  If there is no severity label common to the group of alerts, then default to `info`.  That looks like `or .GroupLabels.severity "info"` in Go templating.  To properly escape for Helm templating, I needed to have the Helm templating engine output the literal string `{{`, since Alertmanager templating syntax is the same as Helm.  To do that, I use the expression `"{{"` inside the double bracket syntax for evaluating Go template expressions.  Thus `{{ "{{" }}`.

Comment (stehessel) [reply dp-terraform/helm/rhacs-terraform/charts/observability/templates/01-operator-03-secret-alertmanager.yaml:23]: Alright thanks for the explanation. Can you please put this explanation as a comment above this line?

**Comment by stehessel** [comment]
I guess it'll only be rolled out for stage anyway, as we would need a new release to rollout the prod changes.
